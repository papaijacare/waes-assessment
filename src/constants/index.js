export const COLORS = {
  RED: 'red',
  YELLOW: 'yellow',
  GREEN: 'green',
}

export const ROLES = {
  MARKING: 'ROLE_MARKING',
  FILTERING: 'ROLE_FILTERING',
}
