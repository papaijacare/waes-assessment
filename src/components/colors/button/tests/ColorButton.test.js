import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import configureStore from 'redux-mock-store'

import { Button } from 'semantic-ui-react'
import { ColorButton } from '../..'
import { ROLES, COLORS } from '../../../../constants'

const mockStore = configureStore()
const store = mockStore({
  colors: {
    filteredColors: [],
    selectedColor: undefined,
  },
})

const instance = () => {
  const props = {
    store,
    role: ROLES.FILTERING,
    color: COLORS.RED,
  }
  const wrapper = shallow(<ColorButton {...props} />)

  return {
    wrapper,
  }
}

describe('ColorButton', () => {
  test('renders correctly', () => {
    const { wrapper } = instance()
    const component = wrapper.dive()

    expect(toJson(component)).toMatchSnapshot()
  })

  test('onClick', () => {
    const { wrapper } = instance()
    const component = wrapper.dive()

    component.find(Button).props().onClick()

    expect(store.getActions()).toMatchSnapshot()
  })
})
