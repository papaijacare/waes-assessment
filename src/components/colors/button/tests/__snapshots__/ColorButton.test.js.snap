// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorButton onClick 1`] = `
Array [
  Object {
    "payload": "red",
    "type": "SET_FILTERED_COLOR",
  },
]
`

exports[`ColorButton renders correctly 1`] = `
<Button
  as="button"
  className=""
  color="red"
  onClick={[Function]}
  role="button"
>
  RED
</Button>
`
